bundler/spec/bundler/resolver/candidate_spec.rb

```diff
@@ -3,8 +3,8 @@
 RSpec.describe Bundler::Resolver::Candidate do
   it "compares fine" do
     version1 = described_class.new("1.12.5", :specs => [Gem::Specification.new("foo", "1.12.5") {|s| s.platform = Gem::Platform::RUBY }])
-    version2 = described_class.new("1.12.5")
+    version2 = described_class.new("1.12.5", :specs => [Gem::Specification.new("foo", "1.12.5") {|s| s.platform = Gem::Platform::X64_LINUX }])
```

Comment:
Is there any reason to replace this spec instead of keeping this one and adding new one 🤔 ?

Reply:
The spec as written previously no longer works, the result is reversed.

The explanation for that is that previously comparison would take into account whether associated specs had been given originally when instantiating the candidate or not. I realized that this did not make sense and was incorrect. Other than the version, the comparison should only take into account whether the candidate is platform specific or not (platform specific candidates should sort higher than generic candidates).

Because of changing the above logic, the result is reversed because passing `:specs => []` now essentially means creating a platform specific candidate, because the platforms of the passed specs (empty) are not all `Gem::Platform::RUBY`. And platform specific candidates sort higher.

But the above felt just an implementation detail, the resolver in practice either instantiates candidates by passing associated `:specs` to `initialize` or by explicitly making them "generic" or "platform specific" using `generic!` or `platform_specific!` after instantiation.

That's why I replaced the spec instead of adding a new one. But I restored the original spec now (changing the expectation) and added a few more.

Reply:
Thanks for detailed explanation.

Reply:
Great, I'm now preparing an upstream patch for pub_grub.